pkg/argocd/applications.go

```diff
@@ -198,34 +195,18 @@ func (builder *ApplicationBuilder) Create() (*ApplicationBuilder, error) {
 	glog.V(100).Infof("Creating argocd application %s in namespace: %s", builder.Definition.Name,
 		builder.Definition.Namespace)
 
-	var err error
-	if !builder.Exists() {
-		unstructuredApplication, err := runtime.DefaultUnstructuredConverter.ToUnstructured(builder.Definition)
-
-		if err != nil {
-			glog.V(100).Infof("Failed to convert structured Application to unstructured object")
-
-			return nil, err
-		}
-
-		unsObject, err := builder.apiClient.Resource(
-			GetApplicationsGVR()).Namespace(builder.Definition.Namespace).Create(
-			context.TODO(), &unstructured.Unstructured{Object: unstructuredApplication}, metav1.CreateOptions{})
-
-		if err != nil {
-			glog.V(100).Infof("Failed to create Application")
-
-			return nil, err
-		}
-
-		builder.Object, err = builder.convertToStructured(unsObject)
+	if builder.Exists() {
+		return builder, nil
+	}
 
-		if err != nil {
-			return nil, err
-		}
+	err := builder.apiClient.Create(context.TODO(), builder.Definition)
+	if err != nil {
+		return nil, err
 	}
 
-	return builder, err
+	builder.Object = builder.Definition
+
+	return builder, nil
```

Comment:
Instead of having multiple returns, you can do something like this:
```
	if !builder.Exists() {
		err = builder.apiClient.Create(context.TODO(), builder.Definition)
		if err == nil {
			builder.Object = builder.Definition
		}
	}

	return builder, err
```